File: Gixor-Backend/Gixor-Api/Gixor-Api-GitHub/src/main/java/com/cpy3f2/Gixor/Domain/DTO/ForkDTO.java
package com.cpy3f2.Gixor.Domain.DTO;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author : simon
 * @description :
 * @last : 2024-11-05 19:28
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-05 19:28
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ForkDTO {
    private String organization;
    private String name;
    @JsonProperty("default_branch_only")
    private Boolean defaultBranchOnly;
}

File: Gixor-Mobile/app/src/main/java/com/cpy3f2/gixor_mobile/model/entity/ForkDTO.kt
package com.cpy3f2.Gixor.Domain.DTO

import com.google.gson.annotations.SerializedName;

/**
 * @author : simon
 * @description :
 * @last : 2024-11-05 19:28
 * Copyright (c) 2024. 保留所有权利。
 * @since : 2024-11-05 19:28
 */

data class ForkDTO(
    @SerializedName("organization")
    val organization: String,
    @SerializedName("repository")
    val name: String,
    @SerializedName("defaultBranchOnly")
    val defaultBranchOnly: Boolean? = null
)
